bcgov/HOUS-permit-portal
# Save PDFs as supporting documents on submission

## Description

On submission of a permit application, the application form and step code checklist form should be converted into PDFs and saved as supporting documents. The backend PDF download does not support react context so the checklist PDF components now take in the checklist as props.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-943](https://hous-bssb.atlassian.net/browse/BPHH-943)

## Steps to QA

1. Create a new permit application and upload a step code checklist using the compliance tool. Submit the application.
2. Login as a reviewer for the application's jurisdiction and open the submitted application
3. Open the "Download application" modal and verify the application PDF & checklist PDF can be downloaded either individually or as part of the zip file.
4. Do the same but do not use the compliance tool for the checklist and instead upload a PDF using the file upload option. Now the PDF should appear in the "Download application" modal and the zip file.

## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

Build vite SSR? @taw27 @sunto?
